KSPCommunityFixes/Performance/OptimisedVectorLines.cs

```diff
@@ -0,0 +1,261 @@
+                m_Viewport.x + (1.0f + clip.x) * m_Viewport.width * 0.5f,
+                m_Viewport.y + (1.0f + clip.y) * m_Viewport.height * 0.5f,
+                clip.z);
+        }
+
+        public static Vector3 WorldToViewportPoint(Camera camera, Vector3 worldPosition)
+        {
+            //if (!patchEnabled)
+            //    return camera.WorldToViewportPoint(worldPosition);
+
+            if (cacheDirty)
+                UpdateCache();
+
+            Vector3 clip = WorldToClip(m_WorldToClip, worldPosition);
+
+            return new Vector3(
+                m_Viewport.x + (1.0f + clip.x) * 0.5f,
+                m_Viewport.y + (1.0f + clip.y) * 0.5f,
+                clip.z);
+        }
+
+        public static Vector3 ScreenToWorldPoint(Camera camera, Vector3 screenPosition)
+        {
+            //if (!patchEnabled)
+            //    return camera.ScreenToWorldPoint(screenPosition);
+
+            if (cacheDirty)
+                UpdateCache();
+
+            // A two multiplication world to clip-space conversion, exactly in reverse.
+            // TODO: I would like if this could be done with one multiplication like WorldToScreenPoint, but I don't understand it well enough.
+
+            // Resources:
+            // https://learnopengl.com/Getting-started/Coordinate-Systems
+            // https://www.songho.ca/math/homogeneous/homogeneous.html
+            // https://stackoverflow.com/questions/7692988/opengl-math-projecting-screen-space-to-world-space-coords
+
+            // Examples:
+            // https://github.com/OpenSAGE/OpenSAGE/blob/master/src/OpenSage.Game/Graphics/ViewportExtensions.cs#L46
+            // https://github.com/rickomax/psxprev/blob/no-deps/Common/GeomMath.cs#L248
+            // https://github.com/Prograda/Skybolt/blob/b9d4f410ecfe4e2ba1634971a0ecd017815192db/src/Skybolt/SkyboltQt/Viewport/ScreenTransformUtil.cpp#L36
+            // https://github.com/GrognardsFromHell/OpenTemple/blob/master/Core/GFX/NumericsExtensions.cs#L40
+
+            // It must replicate the behaviour of Camera.ScreenToWorldPoint exactly, including the third component of screenPosition being in world units.
+            // https://docs.unity3d.com/ScriptReference/Camera.ScreenToWorldPoint.html
+
+            // Normalized device coordinates.
+            Vector4 clipSpace = new Vector4()
+            {
+                x = (screenPosition.x - m_Viewport.x) / m_Viewport.width * 2f - 1f,
+                y = (screenPosition.y - m_Viewport.y) / m_Viewport.height * 2f - 1f,
+                z = 1f,
+                w = screenPosition.z,
+            };
+
+            // Perspective multiplication moves it to clip space.
+            clipSpace.x *= clipSpace.w;
+            clipSpace.y *= clipSpace.w;
+            clipSpace.z *= clipSpace.w;
+
+            Vector4 viewSpace = m_projectionInv * clipSpace;
+            viewSpace.w = 1f;
+            Vector4 worldSpace = m_worldToCameraInv * viewSpace;
```

Comment:
Related to the above comment about using a double backed 4x4 matrix, I would recommend using a vector4d (passing it by ref) here to avoid intermediary float to double conversions, and mutating a single vector4d it instead of instantiating new ones. 

Instantiating new structs for intermediary results is actually pretty expensive.